packages/tools/src/tools/VolumeCroppingTool.ts

```diff
@@ -0,0 +1,1534 @@
+    if (sphereState) {
+      return;
+    }
+    const sphereSource = vtkSphereSource.newInstance();
+    sphereSource.setCenter(point);
+    const sphereRadius =
+      this.configuration.sphereRadius !== undefined
+        ? this.configuration.sphereRadius
+        : 15;
+    sphereSource.setRadius(sphereRadius);
+    const sphereMapper = vtkMapper.newInstance();
+    sphereMapper.setInputConnection(sphereSource.getOutputPort());
+    const sphereActor = vtkActor.newInstance();
+    sphereActor.setMapper(sphereMapper);
+    let color: [number, number, number] = [0.0, 1.0, 0.0]; // Default green
+    const sphereColors = this.configuration.sphereColors || {};
+
+    if (cornerKey) {
+      color = sphereColors.corners || [0.0, 0.0, 1.0]; // Use corners color from config, fallback to blue
+    } else if (axis === 'z') {
+      color = sphereColors.z || [1.0, 0.0, 0.0];
+    } else if (axis === 'x') {
+      color = sphereColors.x || [1.0, 1.0, 0.0];
+    } else if (axis === 'y') {
+      color = sphereColors.y || [0.0, 1.0, 0.0];
+    }
+    // Store or update the sphere position
+    const idx = this.sphereStates.findIndex((s) => s.uid === uid);
+    if (idx === -1) {
+      this.sphereStates.push({
+        point: point.slice(),
+        axis,
+        uid,
+        sphereSource,
+        sphereActor,
+        isCorner: !!cornerKey,
+        color,
+      });
+    } else {
+      this.sphereStates[idx].point = point.slice();
+      this.sphereStates[idx].sphereSource = sphereSource;
+    }
+
+    // Remove existing actor with this UID if present
+    const existingActors = viewport.getActors();
+    const existing = existingActors.find((a) => a.uid === uid);
+    if (existing) {
+      //viewport.removeActor(uid);
+      return;
+    }
+    sphereActor.getProperty().setColor(color);
+    sphereActor.setPickable(true);
+    viewport.addActor({ actor: sphereActor, uid: uid });
+  }
+
+  _initialize3DViewports = (viewportsInfo): void => {
+    const [viewport3D] = viewportsInfo;
+    const renderingEngine = getRenderingEngine(viewport3D.renderingEngineId);
+    const viewport = renderingEngine.getViewport(viewport3D.viewportId);
+    const volumeActors = viewport.getActors();
+    const imageData = volumeActors[0].actor.getMapper().getInputData();
+    const dimensions = imageData.getDimensions();
+    const origin = imageData.getOrigin();
+    const spacing = imageData.getSpacing(); // [xSpacing, ySpacing, zSpacing]
+    const cropFactor = 0.2;
+    const xMin = origin[0] + cropFactor * (dimensions[0] - 1) * spacing[0];
+    const xMax =
+      origin[0] + (1 - cropFactor) * (dimensions[0] - 1) * spacing[0];
+    const yMin = origin[1] + cropFactor * (dimensions[1] - 1) * spacing[1];
+    const yMax =
+      origin[1] + (1 - cropFactor) * (dimensions[1] - 1) * spacing[1];
+    const zMin = origin[2] + cropFactor * (dimensions[2] - 1) * spacing[2];
+    const zMax = origin[2] + 0.8 * (dimensions[2] - 1) * spacing[2];
```

Comment:
You use zMax = origin[2] + 0.8 * (dimensions[2] - 1) * spacing[2];
This is inconsistent with the other axes, which use (1 - cropFactor).
If this is intentional, that's fine, but if not, it should be: ```    const zMax = origin[2] + (1 - cropFactor) * (dimensions[2] - 1) * spacing[2];```